sql/src/main/java/io/crate/execution/dsl/phases/MergePhase.java

```diff
@@ -59,20 +62,57 @@
     /**
      * expects sorted input and produces sorted output
      */
-    @Nullable private PositionalOrderBy positionalOrderBy;
+    @Nullable
+    private PositionalOrderBy positionalOrderBy;
+
+    public static boolean isNeeded(Collection<String> executionNodes,
+                                   ResultDescription resultDescription,
+                                   boolean isDistributed) {
```

Comment:
same here, the `isDistributed` is currently rather specific to joins - I think we should try to avoid putting it here - especially without any sort of documentation that explains the flag